pkgs/development/python-modules/meep/default.nix

```diff
@@ -0,0 +1,139 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, gfortran
+, mpi
+, blas
+, lapack
+, fftw
+, hdf5-mpi
+, swig
+, gsl
+, harminv
+, libctl
+, libGDSII
+, openssh
+, guile
+, python
+, numpy
+, scipy
+, matplotlib
+, h5py-mpi
+, cython
+, autograd
+, mpi4py
+}:
+
+buildPythonPackage rec {
+  pname = "meep";
+  version = "1.25.0";
+
+  src = fetchFromGitHub {
+    owner = "NanoComp";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-4rIz2RXLSWzZbRuv8d4nidOa0ULYc4QHIdaYrGu1WkI=";
+  };
+
+  format = "other";
+
+  nativeBuildInputs = [
+    autoreconfHook
+    gfortran
+    pkg-config
+    swig
+    mpi
+  ];
+
+  buildInputs = [
+    gsl
+    blas
+    lapack
+    fftw
+    hdf5-mpi
+    harminv
+    libctl
+    libGDSII
+    guile
+    gsl
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+    numpy
+    scipy
+    matplotlib
+    h5py-mpi
+    cython
+    autograd
+    mpi4py
+  ];
+
+  propagatedUserEnvPkgs = [ mpi ];
+
+  dontUseSetuptoolsBuild = true;
+  dontUsePipInstall = true;
+  dontUseSetuptoolsCheck = true;
+
+  enableParallelBuilding = true;
+
+  preConfigure = ''
+    export HDF5_MPI=ON
+    export PYTHON=${python}/bin/${python.executable};
+  '';
+
+  configureFlags = [
+    "--without-libctl"
+    "--enable-shared"
+    "--with-mpi"
+    "--with-openmp"
+    "--enable-maintainer-mode"
+  ];
+
+  passthru = { inherit mpi; };
+
+  doCheck = true;
+  checkPhase = ''
```

Comment:
The results of the run are not really checked here? Right now only two things are verified: does import work and does it run without crashing. Should be maybe remove it or simplify the test?

Reply:
Yeah, that is correct. Then again checking those two is better than not checking them. Only when `sim.run()` is called, MPI is actually used and the MPI runtime errors pop up. Specifically, the undefined references mentioned in the above comment are not found otherwise. This is the basic example from the MEEP manual: https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/
Verifying the results numerically is certainly more difficult as it produces volumetric data. Conclusion; I would not boil it down to something below an actually running simulation or remove it, but if you have a good idea how to also verify the results let me know :)

Reply:
Is there any way we can simplify the part between`import` and `sim.run`?

Reply:
Ok, can you please add the link to the tutorial and a short comment about the purpose of the test.

Reply:
Alright, has been done.